state_log/src/tests.rs

```diff
@@ -0,0 +1,56 @@
+use core::time;
+use r2d2_redis::redis::Commands;
+use r2d2_redis::redis::streams::StreamRangeReply;
+use r2d2_redis::{RedisConnectionManager, r2d2::Pool};
+use testcontainers::{
+  GenericImage, ImageExt,
+  core::{IntoContainerPort, WaitFor},
+  runners::AsyncRunner,
+};
+use tokio;
+
+use crate::logger::Sender;
+use crate::logger::log;
+use crate::logger::log_event_sender;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_redis() {
+  let container = GenericImage::new("redis", "8.0.2-alpine")
+    .with_exposed_port(6379.tcp())
+    .with_wait_for(WaitFor::message_on_stdout("Ready to accept connections"))
```

Comment:
This is great, but a) needs a comment that this is idiomatic (because it is!), and b) what do you think of adding `.with_wait_for(WaitFor::Healthcheck)`?